content/general/concepts/abstract-factory/abstract-factory.md

```diff
@@ -0,0 +1,396 @@
+
+```java
+public class CreditAgencyGateway {
+  // Depending on which name is searched for, a different customer object is returned
+  public Customer getCustomer(String name) {
+    return switch (name) {
+      case "Harry" -> new Customer("Harry",
+                                   "Potter",
+                                   LocalDate.of(1980, 7, 31),
+                                   "4 Privet Drive, Little Whinging",
+                                   795);
+      case "Ron" -> new Customer("Ron",
+                                 "Weasley",
+                                 LocalDate.of(1980, 3, 1),
+                                 "Shell Cottage, Cornwall",
+                                 379);
+      case "Hermione" -> new Customer("Hermione",
+                                      "Granger",
+                                      LocalDate.of(1979, 9, 19),
+                                      "12 Grimmauld Palace, London",
+                                      843);
+      default -> throw new IllegalArgumentException("Could not return the credit history for " + name);
+    };
+  }
+}
+```
+
+The `CreditAgencyGateway` class uses a `switch` statement to query a provided `name`. If the name is recognized a new customer is returned. Otherwise, an exception is thrown.
+
+One of the main advantages of factory patterns is they allow for a large amount of model classes and enforce a common interface between them. In the code below, an abstract model `CurrentAccount` class works as an extension for its concrete sub-classes to be based on:
+
+```java
+public abstract class CurrentAccount {
+  private final String accountUID = UUID.randomUUID().toString();
+  private final Customer accountHolder;
+  private int balance;
+  private int agreedOverdraftLimit;
+
+  public CurrentAccount(Customer accountHolder, int initBalance, int agreedOverdraftLimit) {
+    this.accountHolder = accountHolder;
+    this.balance = initBalance;
+    this.agreedOverdraftLimit = agreedOverdraftLimit;
+  }
+
+  public void addFunds(final int amount) {
+    balance += amount;
+  }
+
+  public void withdrawFunds(final int amount) {
+    if (amount < balance + agreedOverdraftLimit) {
+      balance -= amount;
+    } else {
+      System.out.println("Insufficient funds. Unable to withdraw " + amount);
+    }
+  }
+
+  // Enforces concrete classes to override this method
+  public abstract void increaseOverdraft(int requestedLimit);
+
+  // Getters
+}
+```
+
+**Note:** Getters have been excluded from the above code snippet.
+
+The parent `CurrentAccount` class above, as well as providing the common fields and constructor, requires its children to implement the `.increaseOverdraft()` method. This is the differentiating feature described between each current account.
+
+The code snippets below are the concrete implementations of `CurrentAccount`. The first is the Gold-level:
+
+```java
+public class GoldPersonal extends CurrentAccount {
```

Comment:
```suggestion
public class GoldPersonal extends CurrentAccount {
```

Did you mean GoldPersonnel or GoldPersonal?

Reply:
I beliew I am correct, and will resolve. Feel free to re open should you dis-agree